github/orgs/retrieve_github_data.py

```diff
@@ -0,0 +1,364 @@
+    endpoint: Any = None, orgs: List[str] = None, all_permitted: bool = False
+) -> List[OrgInfo]:
+    """Generator of org data."""
+    if not endpoint:
+        # if we're creating the endpoint, then arguments must already be
+        # in environment variables.
+        endpoint = get_connection(DEFAULT_GRAPHQL_ENDPOINT, os.environ.get("GH_TOKEN"))
+    if not orgs:
+        if all_permitted:
+            orgs = _all_owned_orgs(endpoint)
+        else:
+            # Just get the ones we are configured to monitor
+            orgs = _orgs_to_check()
+    for org in orgs:
+        data = get_org_info(endpoint, org)
+        yield data
+
+
+# helper classes for graph errors
+def _compact_fmt(d):
+    s = []
+    for k, v in d.items():
+        if isinstance(v, dict):
+            v = _compact_fmt(v)
+        elif isinstance(v, (list, tuple)):
+            lst = []
+            for e in v:
+                if isinstance(e, dict):
+                    lst.append(_compact_fmt(e))
+                else:
+                    lst.append(repr(e))
+            s.append("{}=[{}]".format(k, ", ".join(lst)))
+            continue
+        s.append(f"{k}={v!r}")
+    return "(" + ", ".join(s) + ")"
+
+
+def _report_download_errors(errors):
+    """error handling for graphql comms."""
+    logger.error("Document contain %d errors", len(errors))
+    for i, e in enumerate(errors):
+        msg = e.pop("message")
+        extra = ""
+        if e:
+            extra = " %s" % _compact_fmt(e)
+        logger.error("Error #%d: %s%s", i + 1, msg, extra)
+
+
+def get_connection(base_url: str, token: Optional[str]) -> Any:
+    # corner case -- we come through here during CI when there is no
+    # token and doctests are running, so token is None
+    endpoint = HTTPEndpoint(base_url, {"Authorization": "bearer " + (token or ""),})
+    endpoint.report_download_errors = _report_download_errors
+
+    # determine which date we're collecting for
+    global _collection_date
+    assert _collection_date == "1970-01-01"
+    _collection_date = date.today().isoformat()
+    return endpoint
+
+
+def main() -> None:
+    # hack to support doctests
+    if "pytest" in sys.modules:
+        return
+    args = parse_args()
+    if args.output:
+        csv_out = csv.writer(open(args.output, "w"))
+    else:
+        csv_out = csv.writer(sys.stdout)
+    endpoint = get_connection(args.graphql_endpoint, args.token)
+    if not args.no_csv:
+        csv_out.writerow(OrgInfo.csv_header())
+        for row in get_all_org_data(endpoint, args.orgs, args.all):
+            csv_output(row, csv_writer=csv_out)
+    if not args.no_json:
+        with args.json as jf:
+            for row in get_all_org_data(endpoint, args.orgs, args.all):
+                jf.write(f"{json.dumps(row.as_dict())}\n")
+
+    ## csv_out.writerow(OrgInfo.csv_header())
+    ## for org in args.orgs:
+    ##     row_data = get_org_info(endpoint, org)
+    ##     csv_output(row_data, csv_writer=csv_out)
```

Comment:
should this be removed or added behind a new cli opt?

Reply:
Good question -- probably the cli prompt (TIL).

The intent is to display a more human readable format when run from the cli